mlflow/server/handlers.py

```diff
@@ -1363,6 +1364,25 @@ def _delete_registered_model_tag():
     return _wrap_response(DeleteRegisteredModelTag.Response())
 
 
+def _validate_source(source: str, run_id: str) -> None:
+    if not is_local_uri(source):
+        return
+
+    if run_id:
+        store = _get_tracking_store()
+        run = store.get_run(run_id)
+        source = pathlib.Path(local_file_uri_to_path(source))
+        run_artifact_dir = pathlib.Path(local_file_uri_to_path(run.info.artifact_uri))
+        if run_artifact_dir in [source, *source.parents]:
```

Comment:
If `run_artifact_dir` isn't normalized, this seems like it might not work. For example:

```
In [1]: import pathlib

In [2]: p = pathlib.Path.cwd()

In [3]: p2 = pathlib.PosixPath("/private/tmp/dir1")

In [4]: p
Out[4]: PosixPath('/private/tmp/dir1/dir2/dir3')

In [5]: p2 in [*p.parents]
Out[5]: True

In [6]: p3 = pathlib.PosixPath("/private/tmp/dir1/dir2/..")

In [7]: p3 in [*p.parents]
Out[7]: False
```

Reply:
fixed!